src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -917,704 +988,529 @@ static void WindowFootpathSetProvisionalPathAtPoint(const ScreenCoordsXY& screen
-    int32_t turnedRotation = _footpathConstructDirection - currentRotation + (currentRotation % 2 == 1 ? -1 : 1);
-    WindowFootpathMousedownDirection(turnedRotation);
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
+    window->KeyboardShortcutTurnRight();
 }
 
 void window_footpath_keyboard_shortcut_slope_down()
 {
-    rct_window* w = window_find_by_class(WC_FOOTPATH);
-    if (w == nullptr || WidgetIsDisabled(w, WIDX_SLOPEDOWN) || WidgetIsDisabled(w, WIDX_LEVEL)
-        || WidgetIsDisabled(w, WIDX_SLOPEUP) || w->widgets[WIDX_LEVEL].type == WindowWidgetType::Empty)
-    {
-        return;
-    }
-
-    switch (gFootpathConstructSlope)
-    {
-        case 0:
-            window_event_mouse_down_call(w, WIDX_SLOPEDOWN);
-            break;
-        case 2:
-            window_event_mouse_down_call(w, WIDX_LEVEL);
-            break;
-        default:
-        case 6:
-            return;
-    }
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
+    window->KeyboardShortcutSlopeDown();
 }
 
 void window_footpath_keyboard_shortcut_slope_up()
 {
-    rct_window* w = window_find_by_class(WC_FOOTPATH);
-    if (w == nullptr || WidgetIsDisabled(w, WIDX_SLOPEDOWN) || WidgetIsDisabled(w, WIDX_LEVEL)
-        || WidgetIsDisabled(w, WIDX_SLOPEUP) || w->widgets[WIDX_LEVEL].type == WindowWidgetType::Empty)
-    {
-        return;
-    }
-
-    switch (gFootpathConstructSlope)
-    {
-        case 6:
-            window_event_mouse_down_call(w, WIDX_LEVEL);
-            break;
-        case 0:
-            window_event_mouse_down_call(w, WIDX_SLOPEUP);
-            break;
-        default:
-        case 2:
-            return;
-    }
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
+    window->KeyboardShortcutSlopeUp();
 }
 
 void window_footpath_keyboard_shortcut_demolish_current()
 {
-    rct_window* w = window_find_by_class(WC_FOOTPATH);
-    if (w == nullptr || WidgetIsDisabled(w, WIDX_REMOVE) || w->widgets[WIDX_REMOVE].type == WindowWidgetType::Empty
-        || (!gCheatsBuildInPauseMode && game_is_paused()))
-    {
-        return;
-    }
-
-    WindowFootpathRemove();
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
+    window->KeyboardShortcutDemolishCurrent();
 }
 
 void window_footpath_keyboard_shortcut_build_current()
 {
-    rct_window* w = window_find_by_class(WC_FOOTPATH);
-    if (w == nullptr || WidgetIsDisabled(w, WIDX_CONSTRUCT) || w->widgets[WIDX_CONSTRUCT].type == WindowWidgetType::Empty)
-    {
-        return;
-    }
-
-    window_event_mouse_up_call(w, WIDX_CONSTRUCT);
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
+    window->KeyboardShortcutBuildCurrent();
 }
 
 void WindowFootpathResetSelectedPath()
 {
-    gFootpathSelection = {};
+    auto* window = static_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
```

Comment:
All of the keyboard shortcuts and `WindowFootpathResetSelectedPath` helper functions should perform a nullptr check on the result of window_find_by_class as its possible the shortcut has been pressed and the window is not open.